src/visualizations/config/adapters/dhis_highcharts/legend.js

```diff
@@ -64,7 +63,72 @@ function getLegend(fontStyle, dashboard, visType) {
     )
 }
 
-export default function (isHidden, fontStyle, visType, dashboard) {
+const getLegendSetByDisplayStrategy = ({
+    displayStrategy,
+    legendSets,
+    legendSetId,
+}) => {
+    if (
+        displayStrategy === LEGEND_DISPLAY_STRATEGY_FIXED &&
+        legendSets.length
+    ) {
+        return legendSets[0]
+    } else if (displayStrategy === LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM) {
+        return legendSets.find(legendSet => legendSet.id === legendSetId)
+    } else {
+        return null
+    }
+}
+
+const getBulletStyleByFontStyle = fontStyle =>
+    `display: inline-block; border-radius: 50%; width: ${fontStyle[FONT_STYLE_OPTION_FONT_SIZE]}px; height: ${fontStyle[FONT_STYLE_OPTION_FONT_SIZE]}px; margin-bottom: 4px;`
+
+const formatLabel = ({
+    seriesId,
+    metaData,
+    displayStrategy,
+    legendSets,
+    fontStyle,
+    seriesColor,
+    seriesName,
+}) => {
+    const legendSet = getLegendSetByDisplayStrategy({
+        displayStrategy,
+        legendSets,
+        legendSetId: metaData[seriesId]?.legendSet,
+    })
+
+    let result = '<div style="display: flex; align-items: center;">'
```

Comment:
Perhaps a more readable way is to use an array where tags are pushed and it's joined before returning.

Reply:
True, it was very messy. Updated with an array now thats hopefully easier on the eyes 🙂